web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php

```diff
@@ -64,7 +64,7 @@ public static function create(ContainerInterface $container) {
    * @param \Drupal\dpl_library_agency\Branch\Branch[] $branches
    *   The branches to build the string with.
    */
-  protected function buildBranchesJsonProp(array $branches) : string {
+  public static function buildBranchesJsonProp(array $branches) : string {
```

Comment:
Please reconsider this approach.

Exposing methods on a controller to be used elsewhere does not help separation of concerns. Our intention was originally to let all aps use this controller. If we diverge from this to domain specific controllers then we need a better way to do code sharing.

I see three options:

1. Shared services which can be injected
2. Inheritance from a base controller for serving React application(s)
3. Traits

I would prefer 1.

Reply:
It's not the best solution, but it's what we got right now. It's the first time i hear the plan about this controller.

I think 1. is  correct but it is currently out-of-scope for the time slot given for this assignment.

Reply:
I agree in the first option.
If you don't have the time left for refactoring it into a service, then at least we should have a TODO in the doc block, I think.